ldraw/src/library.rs

```diff
@@ -381,3 +381,46 @@ where
             .collect::<HashMap<_, _>>(),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{collections::HashMap, sync::Arc};
+
+    use crate::{PartAlias, document::{MultipartDocument, Document, BfcCertification}};
+    use super::{PartCache, PartKind};
+
+    #[test]
+    fn test_part_cache_query_existing() {
+        let document = MultipartDocument {
+            body: Document {
+                name: "Doc".to_string(),
+                author: "Author".to_string(),
+                description: "Description".to_string(),
+                bfc: BfcCertification::NoCertify,
+                headers: vec![],
+                commands: vec![],
+            },
+            subparts: HashMap::new(),
+        };
+
+        let mut cache = PartCache::new();
+
+        let existing_key = PartAlias::from("existing".to_string());
+        let document = Arc::new(document);
+
+        cache.register(PartKind::Primitive, existing_key.clone(), Arc::clone(&document));
```

Comment:
You can use `document.clone()` instead of `Arc::clone(&document)`

Reply:
Thanks! Applied in 2e2b213 👍